pxd.c

```diff
@@ -1134,24 +1134,13 @@ static int pxd_init_disk(struct pxd_device *pxd_dev, struct pxd_add_ext_out *add
 {
 	struct gendisk *disk;
 	struct request_queue *q;
+#ifdef __PX_BLKMQ__
 	int err = 0;
+#endif
 
 	if (add->queue_depth < 0 || add->queue_depth > PXD_MAX_QDEPTH)
 		return -EINVAL;
 
-	/* Create gendisk info. */
-	disk = alloc_disk(1);
-	if (!disk)
-		return -ENOMEM;
-
-	snprintf(disk->disk_name, sizeof(disk->disk_name),
-			PXD_DEV"%llu", pxd_dev->dev_id);
-	disk->major = pxd_dev->major;
-	disk->first_minor = pxd_dev->minor;
-	disk->flags |= GENHD_FL_EXT_DEVT | GENHD_FL_NO_PART_SCAN;
-	disk->fops = &pxd_bd_ops;
-	disk->private_data = pxd_dev;
-
```

Comment:
given disk is not initialized, below L1152 wherein disk->fops will fail, if pxd is compiled with MAKEREQ macro (old ways of doing block IO). Actually take the above block under 
#ifdef __PXD_BIO_MAKEREQ_
/// alloc_disk L1142->L1153 above
#endif

We only support PXD_BIO_BLKMQ always.